### PR TITLE
Feature: Live edit .vue files during development

### PR DESCRIPTION
**Pull request by RCK-DEV**
I've added a feature in which vuepress-jsdoc only regenerates the Vue source files which are modified during development (`yarn dev`). 

To run in live-edit mode execute `yarn run dev-live`

For the live edit feature i've used the current code inside the generate function (_cmds/index.js_) and extracted it into separate functions. Thus not altering the original generate code for now.

When the commander executes the generate functions it checks if the `--live` flag is set and determines if it must generate all docs or only which are currently modified.

My decision not to alter the original generate function results into a lot of duplicate code, but i would eventually like to refactor everything inside the `index.js` file so that the newly extracted functions (e.g. `parseFrontmatter, generateVueMdFileData`, ..) will be utilised. I believe this will improve the overall code quality.

### Know issues:

- `UnhandledPromiseRejectionWarning` caused whenever `.vue` files contains invalid syntax, a new `.vue` files gets created or deleted. Requires rerunning `yarn dev-live`.
- When adding or removing `.vue` file side will not get updated.

### CODE REVIEW

**Comment by ph1p** [comment]
Thanks @RCK-DEV ! 
Really like this feature and can't wait to test it (:
I only have one small comment and that is that this project currently uses npm. Could you change it to npm, as there is now a deleted package.lock and an added yarn.lock in the changes?

**Comment by RCK-DEV** [comment]
Ofcourse! I'll change it to npm. Glad to hear. 😃 I'm currently also working on some other features at the moment. I'll keep you posted. 👍🏻🙃

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "vuepress-jsdoc",
-  "version": "3.4.1",
+  "name": "@rck-dev/vuepress-jsdoc",
```
Comment (ph1p) [inline package.json:2]: Please leave it as it was. It's ok if you publish your own version, but it should go out here.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+  "version": "3.6.2",
```
Comment (ph1p) [inline package.json]: I'm ok with a bump to 3.5.0

File: package.json [hunk 3]
```diff
@@ -24,7 +24,7 @@
   "license": "MIT",
   "author": "Philip Stapelfeldt <me@ph1p.dev>",
   "repository": {
-    "url": "https://github.com/ph1p/vuepress-jsdoc.git",
+    "url": "https://github.com/RCK-DEV/vuepress-jsdoc#feature/multi-nav-sidebar",
```
Comment (ph1p) [inline package.json:27]: Please leave it as it was.

File: package.json [hunk 4]
```diff
@@ -76,6 +77,7 @@
   },
   "devDependencies": {
     "@release-it/conventional-changelog": "^2.0.0",
+    "chokidar": "^3.5.1",
```
Comment (ph1p) [inline package.json:80]: Please move this up to the dependencies.

File: example/package.json [hunk 5]
```diff
@@ -4,8 +4,12 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "docs": "../bin/vuepress-jsdoc.js -c ./jsdoc.json --source=./src --dist=./documentation --title=API --exclude=**/*/*.test.js,class.js --partials=./partials/*.hbs",
+    "docs": "../bin/vuepress-jsdoc.js -c ./jsdoc.json --source=./src --dist=./documentation --title=API --exclude=**/{*.test.js,__tests__/**},class.js,__tests__ --partials=./partials/*.hbs",
+    "docs-live": "yarn docs --live",
+    "docs-multi": "yarn docs --multinav",
+    "docs-multi-mono": "yarn docs --multinav --monorepo",
```
Comment (ph1p) [inline example/package.json:10]: Please use `npm run`, to keep it consistent

**Review by ph1p — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (RCK-DEV) [reply package.json:27]: This wasn't meant to be PR'ed. My bad. I've placed a fork on NPM only for development purposes. I'll put it back. 👍🏻

Comment (ph1p) [reply package.json:27]: All good (: I'll do a final review tomorrow and then hopefully merge everything. Good job!

**Comment by ph1p** [comment]
Oh ok I totally misunderstood the PR. I thought that this is already ready for review, but there are still a few "TODO" or "WIP" places in it. 😅

Keep up the great work! 

A little tip from me. The PR is already a bit big and has apparently a few more features than just the live update of the content. Try to split it up a little bit. It is not bad if there are several PRs (:

And try out the CLI by including the module locally (https://docs.npmjs.com/cli/v7/commands/npm-link)

**Comment by RCK-DEV** [comment]
I believe i did another PR by accident a few days ago. 🙈 My bad haha. If you could decline this one than i'll redo my PR's when I'm ready. And in smaller chunks like you said. 😃

**Comment by ph1p** [comment]
All good (: Take your time and play around with the code. I'll close this one and look forward to your changes.

**Pull request closed without merging**